src/Nethermind/Nethermind.Consensus/IBlockProducerRunner.cs

```diff
@@ -0,0 +1,16 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Threading.Tasks;
+using Nethermind.Core;
+
+namespace Nethermind.Consensus;
+
+public interface IBlockProducerRunner
+{
+    Task Start();
+    Task StopAsync();
```

Comment:
We don't have truly async implementations of this?

Reply:
Only clique for stop